libyui-qt-pkg/src/YQPkgHistoryDialog.cc

```diff
@@ -29,277 +29,357 @@
 #include <yui/qt/YQUI.h>
 #include <yui/qt/utf8.h>
+#include <yui/qt/YQSignalBlocker.h>
 
-#include <zypp/parser/HistoryLogReader.h>
 #include <zypp/Date.h>
 #include <zypp/Edition.h>
 #include <boost/ref.hpp>
 
 #include <QApplication>
-#include <QDesktopWidget>
+#include <QBoxLayout>
+#include <QEventLoop>
 #include <QHBoxLayout>
 #include <QLabel>
+#include <QList>
+#include <QMessageBox>
 #include <QPushButton>
 #include <QSplitter>
-#include <QStyle>
-#include <QList>
-#include <QBoxLayout>
 #include <QTreeWidget>
-#include <QMessageBox>
 
 #include "YQPkgHistoryDialog.h"
 #include "YQPkgList.h"
 #include "YQIconPool.h"
 #include "QY2LayoutUtils.h"
 
 
-#define SPACING			2	// between subwidgets
-#define MARGIN			4	// around the widget
+#define SPACING         4	// between subwidgets
+#define MARGIN		9	// around the widget
 
 #define FILENAME	"/var/log/zypp/history"
 
 using std::endl;
 
+
 YQPkgHistoryDialog::YQPkgHistoryDialog( QWidget * parent )
     : QDialog( parent )
 {
     // Dialog title
     setWindowTitle( _( "Package History" ) );
 
-    // Enable dialog resizing even without window manager
     setSizeGripEnabled( true );
+    setMinimumSize( 750, 550 );
+
 
-    // Layout for the dialog (can't simply insert a QVBox)
+    // Outer (main) layout
 
     QVBoxLayout * layout = new QVBoxLayout();
     Q_CHECK_PTR( layout );
-    setLayout(layout);
-    layout->setMargin(MARGIN);
-    layout->setSpacing(SPACING);
+    setLayout( layout );
+    layout->setMargin( MARGIN );
+    layout->setSpacing( SPACING );
+
+
+    // Heading
 
-    setMinimumSize (300,400);
+    QLabel * label = new QLabel(  _( "Package History (/var/log/zypp/history)" ), this );
+    label->setFixedHeight( label->sizeHint().height() );
+    layout->addWidget( label );
 
-    QLabel * label = new QLabel ( _("Show History (/var/log/zypp/history)" ), this );
-    label->setFixedHeight (label->sizeHint ().height ());
-    layout->addWidget (label);
 
-    // VBox for splitter
+    // Splitter between the trees
+
     QSplitter * splitter = new QSplitter( Qt::Horizontal, this );
     Q_CHECK_PTR( splitter );
     layout->addWidget( splitter );
 
-    // History view
-    _dates = new QTreeWidget (splitter);
-    _dates->setColumnCount (1);
-    _dates->setHeaderLabels ( QStringList( _("Date") ) );
 
-    _actions = new QTreeWidget (splitter);
-    _actions->setColumnCount (2);
-    _actions->setHeaderLabels ( QStringList( _("Action") ) << _("Version/URL") );
-    _actions->setColumnWidth (0, 200);
+    // Flat list for the dates
+
+    _datesTree = new QTreeWidget( splitter);
+    _datesTree->setColumnCount( 1 );
+    _datesTree->setHeaderLabels( QStringList( _( "Date" ) ) );
+    _datesTree->setRootIsDecorated( false );
+
+
+    // Tree for the actions: Action items below a date item parent
+
+    _actionsTree = new QTreeWidget( splitter );
+    _actionsTree->setColumnCount( 2 );
+    _actionsTree->setHeaderLabels( QStringList( _( "Action" ) ) << _( "Version/URL" ) );
```

Comment:
Is actually an URL ever displayed when we show only the package actions (install/delete)?